### PR TITLE
Allow for relative paths

### PR DESCRIPTION
**Pull request by hollandben**
Extending from @mbenna's PR at #100 which follows on from issue #91.

This PR allows for relative paths to work without the `baseDir` option interfering.

Also contains some clean up that will mostly be done in #107 

**Note:** I still say that you should be using relative paths from the root for ease, i.e. `/path/to/assets`

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -45,13 +45,12 @@ module.exports = function(grunt) {
                 files: [{
                     expand: true,
                     cwd: 'tmp/',
-                    src: ['*.html', '*.css', '!replace*.*', '!*.php', '!cdnPath.html']
+                    src: ['*.html', '*.css', '!**/replace*.*', '!*.php', '!baseDir.html', '!baseDirOverride.html', '!cdnPath.html', '!deleteOriginals.html']
```
Comment (hollandben) [inline Gruntfile.js]: Needs to be tidied in #107

File: tasks/lib/utils.js [hunk 2]
```diff
@@ -7,6 +7,9 @@ var regexs = require('./regexs');
 
 module.exports = function(opts) {
     return {
+        isRelativePath: function(path) {
```
Comment (hollandben) [inline tasks/lib/utils.js]: Could unit test these. Maybe put in #107 ?

**Comment by hollandben** [comment]
Tests are passing but for some reason Travis is stuck

**Comment by mbenna** [comment]
Yes, I think you nailed the core issue, i.e. that relative paths need different handling than absolute paths when it comes to baseDir.

re: relative vs. absolute... yes, I'd rather use absolute, however sometimes the project is served from root (e.g. http://project.com/index.html) and other times it's served from a subdirectory (e.g. http://testserver.com/projectname/index.html). Adding relative path support to cache busting was easier than getting the whole project always served from root. :)

Thanks again for your help on this. I'm really looking forward to getting these changes into production and therefore getting a much higher % of our assets using far future expiry!

Mike

**Comment by hollandben** [comment]
Completes #100

**Comment by omidraha** [comment]
Hi,
What's the current status of this issue for last version of `grunt-cache-bust`,
I have the same issues when relative path exist on the source,
[Here](https://github.com/omidraha/bugReports/tree/master/grunt-cache-bust/sample-relative-path) is a sample use case for that,
The related path used on this sample use case defined exactly on this [file](https://github.com/omidraha/bugReports/blob/master/grunt-cache-bust/sample-relative-path/assets/js/app/worker.js#L3).
To test, just follow simple command on the Read-Me file.
Then on the generated `dist` directory, you will see that reference of related path is not updated on that JavaScript file.

**Pull request merged**